### ISSUE TITLE
about train fcn questions.

### ISSUE THREAD

**Issue by WZMIAOMIAO** [pytorch/vision#4292]
thanks for your great work!

I have read https://github.com/pytorch/vision/blob/master/references/segmentation/train.py script. And there are some questions.

1. why use aux_classifier for fcn, are there any references ?
2. why is the learning rate of aux_classifier ten times that of base lr?https://github.com/pytorch/vision/blob/master/references/segmentation/train.py#L131
3. If I want to fine-train fcn, how to set the appropriate learning rate? I use default learning rate 0.01 to train PASCAL VOC2012 with res50_rcn,  the result is bad(first epoch, mean IOU=10.1). If not train model, directly evaluate, mean IoU=69.  

cc @datumbox

**Comment by datumbox** [comment]
Good questions.

The FCN is an old algorithm and the [paper](https://arxiv.org/pdf/1411.4038.pdf) came out around the time BatchNorm was invented. So before BN using auxiliaries was a good way to ensure that the bottom parts of the network can be trained effectively (due to vanishing gradient). Adjusting the LR is also a trick to get better results. Setting the hyper parameters requires some hyper-parameter tuning and it's dependent on your problem and data.

I think another question to ask is if you should use FCN or pick one of the latest algos instead. I haven't done huge tests on that and I'm not an expert in semantic segmentation but from [limited](https://pytorch.org/blog/torchvision-mobilenet-v3-implementation/#semantic-segmentation) experiments we see that FCN is often superseded. So perhaps using LR-ASPP will simplify your training and get you marginally better results.

**Comment by WZMIAOMIAO** [comment]
Thank you for your reply. This approach is very similar to inception_v1. In your opinion, if I use resnet as the backbone, I can drop this aux_ classifier. Because resnet have batch normalization module.

**Comment by datumbox** [comment]
Yes but I would also checkout different segmentation heads. If you check the article I linked you will see that FCN was superseded completely by LR-ASPP for the MobileNetV3 case, so we didn't publish the model. Maybe a newer head will yield not only better accuracy but also better speed. :)

**Comment by WZMIAOMIAO** [comment]
Yes, I have check the article. Next, I will learn other segmentation networks in turn. Thank you.


Model | mIoU | Global Pixel Acc | Inference on CPU (sec) | # Params (M)
-- | -- | -- | -- | --
LR-ASPP MobileNetV3-Large | 57.9 | 91.2 | 0.3278 | 3.22
DeepLabV3 MobileNetV3-Large | 60.3 | 91.2 | 0.5869 | 11.03
FCN MobileNetV3-Large (not released) | 57.8 | 90.9 | 0.3702 | 5.05
DeepLabV3 ResNet50 | 66.4 | 92.4 | 6.3531 | 39.64
FCN ResNet50 | 60.5 | 91.4 | 5.0146 | 32.96
